wp-content/themes/borderzine/inc/widgets/class-borderzine-6-col-widget.php

```diff
@@ -0,0 +1,301 @@
+<?php
+
+/**
+ * Borderzine 6-Column recent posts widget and associated functions
+ *
+ * Copied from Largo Recent Posts
+ */
+class Borderzine_6_Col_Widget extends WP_Widget {
+
+	/**
+	 * Register widget with WordPress.
+	 */
+	function __construct() {
+
+		$widget_ops = array(
+			'classname' => 'borderzine-6-col-widget',
+			'description' => __( 'A flexible widget to display recent photo gallery posts.', 'borderzine' )
+		);
+		parent::__construct(
+			'borderzine-6-col-widget', // Base ID
+			__( 'Borderzine 6 Column Widget', 'borderzine' ), // Name
+			$widget_ops // Args
+		);
+
+	}
+
+	/**
+	 * Outputs the content of the recent posts widget.
+	 *
+	 * @param array $args widget arguments.
+	 * @param array $instance saved values from databse.
+	 * @global $post
+	 * @global $shown_ids An array of post IDs already on the page, to avoid duplicating posts
+	 * @global $wp_query Used to get posts on the page not in $shown_ids, to avoid duplicating posts
+	 */
+	function widget( $args, $instance ) {
+
+		global $post,
+			$wp_query, // grab this to copy posts in the main column
+			$shown_ids; // an array of post IDs already on a page so we can avoid duplicating posts;
+		
+		// Preserve global $post
+		$preserve = $post;
+
+		$posts_term = of_get_option( 'posts_term_plural', 'Posts' );
+
+		// Add the link to the title.
+		$title = apply_filters( 'widget_title', empty( $instance['title'] ) ? '' : $instance['title'], $instance, $this->id_base );
+
+		echo $args['before_widget'];
+
+        if ( $title ) echo $args['before_title'] . $title . $args['after_title'];
+        
+        if( ! empty( $instance['linkurl'] ) ) {
+			echo '<p class="morelink btn btn-primary"><a href="' . esc_url( $instance['linkurl'] ) . '">' . esc_html( $instance['linktext'] ) . '</a></p>';
+		}
+
+		$thumb = isset( $instance['thumbnail_display'] ) ? $instance['thumbnail_display'] : 'small';
```

Comment:
Would it make sense to remove the option to control the thumbnail size, and hardcode this to a fixed image size that matches the expected aspect ratio, without using the "Large" image size, which is larger than strictly necessary in this location?

See the comment later in the LESS file about image size choice and distortion.

Reply:
@benlk Removed/updated to `rect_thumb_half` in https://github.com/INN/umbrella-borderzine/pull/75/commits/f766833d9593870549f624102f4c410abd663258.